docs/js-core-3/module-systems/lesson.md

```diff
@@ -0,0 +1,399 @@
+| 3. Run `npm run cowsaying`.                                                              |
+| 4. What happens? Discuss with a classmate what you think happens and why.                |
+
+### Using packages in our apps
+
+Now that we've looked at how NPM installs packages on our computers, how do we get access to them in our apps? Let's take a look at an example:
+
+```js
+const cowsay = require("cowsay");
+```
+
+It's the `require` function again! This is why NPM is so useful: it does all the hard work needed so that we can import the package just like any other file.
+
+You might have noticed that the argument given to `require` is a bit unusual. There's no `./` or `../`. This is because it's not actually a file path!
+
+When we `require` a package, we can use a special syntax that is just the name of the package. Since there are no dots at the start, the module system knows it needs to look for a package and not a file that we've written.
+
+### Installing a new package
+
+We've just seen how we can install packages that are specified for us by someone else. But we can also use NPM to install new packages ourselves!
+
+If we run:
+
+```
+$ npm install my-package
+```
+
+This will do a couple of things:
+
+1. NPM downloads the files, just like we've seen `npm install` do already
+2. Adds `my-package` to the `package.json` file
+
+This means that not only will you able to use the `my-package` package in your code, but if someone clones your repo, then they would also download the package when they run `npm install`. This is really helpful when writing applications in teams: we can guarantee that anyone who clones and installs your repo will have the packages available on their computer.
+
+| Exercise C                                                                                                      |
+| --------------------------------------------------------------------------------------------------------------- |
+| 1. In the repo that we cloned in Exercise B, run `npm install figlet`.                                          |
+| 2. Open up `exercise-c.js` in VSCode. Take 5 minutes to read the code and discuss what it does in small groups. |
+| 3. Add `const figlet = require("figlet");` at the top of the file.                                              |
+| 4. Run `npm run ascii`.                                                                                         |
+
+## Using ECMAScript Modules
+
+In some code that you see on the internet, you may find a different module system being used. Don't panic! It is very similar to the CommonJS system that you just learned about.
+
+This new module system is typically called ECMAScript Modules, or ESM for short.
+
+Why are there 2 module systems? Unlike `require` & `module.exports`, which use bits of JS that use already know (functions, variables and objects), the ESM system introduces some new syntax. ESM is built into the language itself.
```

Comment:
Maybe instead of referring to them primarily as Commonjs vs ESM.. we should refer to them as 'require style' vs 'import style' and mention the official names just as a parenthetical aside - is that a stupid idea, just thinking out loud here -